src/main/java/uk/gov/companieshouse/api/accounts/controller/CompanyAccountController.java

```diff
@@ -19,7 +22,12 @@
 
     @PostMapping(value = "/transactions/{transactionId}/company-accounts",
             produces = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity createCompanyAccount(@Valid @RequestBody CompanyAccount companyAccount) {
-        return ResponseEntity.status(HttpStatus.CREATED).body(companyAccountService.save(companyAccount));
+    public ResponseEntity createCompanyAccount(@Valid @RequestBody CompanyAccount companyAccount,
+            HttpServletRequest request)
+            throws NoSuchAlgorithmException {
+        Transaction transaction = (Transaction) request.getAttribute("transaction");
```

Comment:
"transaction" is being hard coded in multiple places in this PR and https://github.com/companieshouse/company-accounts.api.ch.gov.uk/pull/18. It would be better to place it in a separate class to avoid issues in the future if someone changes it.